earth_enterprise/integration_test/step_impl/genewimageryresource.py

```diff
@@ -0,0 +1,79 @@
+from getgauge.python import step, before_scenario, Messages
+import os, random, fnmatch, subprocess, datetime
+from subprocess import Popen
+import shutil
+
+# ------------------------
+# Helpful Not-Step Methods
+# ------------------------
+def get_env_value(sKey):
+   return os.getenv(sKey, "unset")
+
+def doesKhassetFileExist(sPathUntilDotKiasset):
+   return os.path.exists(sPathUntilDotKiasset + ".kiasset/khasset.xml" )
+
+def executeCommand(sTestDir):
+   sImageryRoot = "/opt/google/share/tutorials/fusion/";
+   sImageryFile = "Imagery/bluemarble_4km.tif"
+   
+   # What if the Imagery wasn't downloaded
+   sCommandToRun = "sudo " + sImageryRoot + "download_tutorial.sh >/dev/null 2>/dev/null"
+   pHandle = subprocess.Popen(sCommandToRun, shell=True)
+   assert (pHandle.wait() == 0)
+   
+   sCommandUnderTest = "/opt/google/bin/genewimageryresource"
+   sCommand = sCommandUnderTest + " -o \"" + sTestDir + "\" " + sImageryRoot + sImageryFile + " 2>/dev/null"
+   pHandle = subprocess.Popen(sCommand, shell=True)
+   # Reason for the wait or poll is: if the process started but it returned FAILURE (example incorrect input, and the executable returned a non-zero response, the wait would have failed.
+   assert ( (pHandle.wait() == 0) or (pHandle.poll() != None) )
+
+
+# ------------------------
+# Helpful ENV settings
+# ------------------------
+HOME = get_env_value("HOME")
+now = datetime.datetime.now()
+outputDirectoryForTest = "genewimageryresourceTestOutput" + "." + now.strftime("%Y.%m.%d.%H.%M.%S")
+outputROOT = "/gevol/assets"
+
+sCommandScript = "./step_impl/balaOutputPathTestDriver.sh "
+
+# ---------------------------
+# The Tests
+# ------------------------
+@step("perform ge new imagery resource simple test")
+def genewimageryresourceSimpleTest() :
```

Comment:
Would it be possible to refactor all these functions or most of them into one accepting parameters as arguments and in the .spec we call the refactored function passing different parameters ? as an example of Gauge test with parameters see [this ](https://github.com/google/earthenterprise/blob/master/earth_enterprise/integration_test/step_impl/assets.py)(line 62)

Reply:
It wasn't effective for this section of positive and negative unit testing